rostime/include/ros/time.h

```diff
@@ -151,7 +151,10 @@ namespace ros
 
     double toSec()  const { return (double)sec + 1e-9*(double)nsec; };
     T& fromSec(double t) {
-      sec = (uint32_t)floor(t);
+      int64_t sec64 = (int64_t)floor(t);
+      if (sec64 < 0 || sec64 > UINT_MAX)
+        throw std::runtime_error("Duration is out of dual 32-bit range");
```

Comment:
Consider changing this to:
`"Time is out of dual unsigned 32-bit range"`.
Avoids ambiguity for people who are trying to debug, and it is more accurate. The ros_comm repo is currently failing the `empty_timestamp_crash_check` test with this exception, which is why I'm making the suggestion.